tools/tpm2_hash.c

```diff
@@ -57,6 +57,11 @@ struct tpm_hash_ctx {
     TSS2_SYS_CONTEXT *sapi_context;
 };
 
+static tpm_hash_ctx ctx = {
+    .hierarchyValue = TPM_RH_NULL,
+    .halg = TPM_ALG_SHA1,
+};
+
```

Comment:
Instead of making this a global variable, what about making the opt and arg callbacks to get a void * parameter so tools can pass a data structure used to keep their state.